### ISSUE TITLE
[Bug] Downloaded files don't appear anywhere file manager or other apps can work with

### ISSUE THREAD

**Issue by joshtriplett** [mozilla-mobile/fenix#15762]
## Steps to reproduce

I'm using the latest version of Fenix on Android. I accessed a music file someone had hosted on their domain. I got the tiny audio-player interface, and could play the file. I wanted to download that file and add it to the music on my device. I long-pressed on the audio-player interface, and got the option to download the file. After a moment, the file showed as downloaded and gave me the option to open it (which I declined, since I didn't want to open it in Firefox). I also have a persistent notification from Firefox about the completed download, which can't be swiped away, and which does nothing when touched.

I opened the Android file manager ("Files"), and searched, and could not find the file by name or by browsing.

I opened Firefox's downloads, and could not find any way to access the file. I could *open* it using an application, but I want to move it to my music folder where my media player can play folders of music.

### Expected behavior

I expected the downloaded file to appear somewhere the Android file manager ("Files") could see and work with.

### Actual behavior

I'm unable to see that the file exists or do anything with it, with anything other than Firefox's downloads list.

### Device information

* Android device: Pixel 3 XL
* Fenix version: 81.1.2

**Comment by evilpie** [comment]
Probably a duplicate of #8754. You can try Nightly and see if it is fixed for you.

**Comment by joshtriplett** [comment]
No, this isn't a duplicate of #8754. As far as I can tell, #8754 is about indexing media files specifically, wherever they happen to be. This issue is about downloads of any kind of file being somewhere the file manager can see, as files, not as "over there there's a music file". I'm glad that Firefox is arranging to index media files, but in this case, my concern is about file downloads in general, and putting them somewhere that the file manager can manage as a file.

In other words, I care where the file actually gets stored, not about whether it's indexed.

**Comment by Amejia481** [comment]
> No, this isn't a duplicate of #8754. As far as I can tell, #8754 is about indexing media files specifically, wherever they happen to be. This issue is about downloads of any kind of file being somewhere the file manager can see, as files, not as "over there there's a music file". I'm glad that Firefox is arranging to index media files, but in this case, my concern is about file downloads in general, and putting them somewhere that the file manager can manage as a file.
> 
> In other words, I care where the file actually gets stored, not about whether it's indexed.

@joshtriplett thanks for reporting!

Could you share you Android OS version? Is it possible to share the link that you were downloading? And finally if it's not much trouble could you try on [nightly ](https://play.google.com/store/apps/details?id=org.mozilla.fenix) and see if the issue is reproducible, thanks in advance.

**Comment by jcosentino** [comment]
I have noticed this problem as well. Whenever I download photos, they don't appear in my Google Photos even though my Downloads sync to it.
Here is a screenshot of a photo downloaded from Fenix:
https://i.imgur.com/9lVbtCf.png

Take note of the location: /Download/Download/...
I do not have a Download folder inside of my Download folder, so maybe it's somehow saving to root?

**Comment by OddEvans** [comment]
I don't want to open an issue as my problem is related to this.

I have 
- Samsung Galaxy S10e SM-G970U1 
- Android 10 / One UI 2.1 / September 1 2020 Security Patch
- Firefox Daylight(Android) V. 81.1.2 (Build #2015767043)

_**I haven't tested**_ with files like pdf, rar, etc.
**I have tested** with images be it jpeg, png, gif and videos mp4 and webm.

The problem is that Firefox downloads the files fine but does not add them(annexed them??) To download folder, what I mean is I download the file it appears in the folder( I check with MiXplorer from XDA ) but when I go to the default Gallery app it does not appear ir the folder of Download even though in the previous version it did, something I found made them appear is moving the files out of the folder and moving them back to the folder that way they appear in the Gallery app.

Please fix this as soon as you can please because this is really troublesome for me personally and I don't feel like using FF until fixed.

**Comment by Amejia481** [comment]
> I don't want to open an issue as my problem is related to this.
> 
> I have
> 
>     * Samsung Galaxy S10e SM-G970U1
> 
>     * Android 10 / One UI 2.1 / September 1 2020 Security Patch
> 
>     * Firefox Daylight(Android) V. 81.1.2 (Build #2015767043)
> 
> 
> _**I haven't tested**_ with files like pdf, rar, etc.
> **I have tested** with images be it jpeg, png, gif and videos mp4 and webm.
> 
> The problem is that Firefox downloads the files fine but does not add them(annexed them??) To download folder, what I mean is I download the file it appears in the folder( I check with MiXplorer from XDA ) but when I go to the default Gallery app it does not appear ir the folder of Download even though in the previous version it did, something I found made them appear is moving the files out of the folder and moving them back to the folder that way they appear in the Gallery app.
> 
> Please fix this as soon as you can please because this is really troublesome for me personally and I don't feel like using FF until fixed.
> 
> Edit: I don't know if this bug I experience is related to #8754 most likely it is so that's why I didn't want to open issue, and see if I can help with this one with my info

Thanks @OddEvans,  the info is helpful to us. If it's not much trouble could you try on [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix) and see if the issue is reproducible, thanks in advance :)

**Comment by KrasnayaPloshchad** [comment]
What happened of you see the folder with X-plore?

**Comment by OddEvans** [comment]
> Thanks @OddEvans, the info is helpful to us. If it's not much trouble could you try on [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix) and see if the issue is reproducible, thanks in advance :)

Alright some more information.

I tried FF Nightly V. Nightly 201012 17:02 (Build #2015769291)
The problem seems to be almost gone bar a few exceptions.

So I tested a variety of websites including R18 ones lol (IDGAF what you think), just wanna help.

_Using Ublock Origin add-on in Nightly_
_Using Ublock Origin, HTTPS Everywhere and Dark Reader in stable_

**Imgur.com** works in Nightly, tried stable and after sometime they appeared in the Gallery app so may be random at least in stable or need time to appear only site this happens.
**Skscans.com** works on Nightly, not in stable, these are images.
**Mangadex.org** works on Nightly, not in stable, images.
**freeimage.host** works on Nightly, not in stable.
**rule34hentai.net** does not works in either, they download but not appear in app.
**rule34.xxx** works in Nightly, not in stable, images/videos.
**twitter.com** works in Nightly, not in stable.

**Some more info** I tried searching the files using the default 'My Files' samsung app if you go to the Download shortcut the files downloaded in stable don't appear those downloaded in Nightly do!. Another thing is you can find the FF stable downloads if you chose to go fist to _Internal Memory and then Download Folder_ they do appear in the folder there.

Another thing is when you download a Video in stable or Nightly if you select **Open** file the first thing it pops is to choose with which app to open it if you don't have selected a default app(I think) anyway in the box there appear to be **two Gallery apps**, I don't know if this is a visual bug or something else.

**Comment by Amejia481** [comment]
>  **rule34hentai.net** does not works in either, they download but not appear in app.

Thank you so much for all the help, we really appreciated!

Let's ask to the QA team to verify too as they have a wide range of devices and can help us to find possibile bugs :)

**Comment by liuche** [comment]
@Amejia481 I actually also see this issue! 😬 

Two things:
- We are putting downloaded files in the `Downloads/Downloads` folder, but on some Android versions (aka before I upgraded to 11) going into the Downloads folder doesn't show another nested Downloads folder (I have no idea why).
- Other apps seem to name their folder by the app name (see screenshot) so maybe we should actually be putting files in `Downloads/Firefox`?

**Comment by ebalazs-sv** [comment]
I had tested this issue on the latest Nightly 10/27, Beta 83.0.0-beta.1 from 10/21 and Release 82.1.1 from 10/21 with 
- Nexus 5 (Android 6.0.1);
- Huawei P9 Lite (Android 7);
- Motorola Moto G6 (Android 8);
- Pixel 2 (Android 9)

and the downloaded files are added to the Downloads/Downloads folder and are visible in the Downloads app folder and in Files - Downloads device folder.

Tested sites:
- https://file-examples.com;
- https://mobile.twitter.com;
- https://edition.cnn.com;
- https://m.youtube.com;
- https://www.xbox.com;
- https://www.nytimes.com;
- https://www.reddit.com;
- https://www.espn.com;

@Amejia481 I will remove the qa:needed label for now. Let me know if other scenarios are required.

**Comment by Amejia481** [comment]
Thanks @ebalazs-sv !

**Comment by OddEvans** [comment]
@Amejia481

Some more test I did in New Firefox Daylight 82.1.1 (Build #2015770923)

● Samsung S10e SM-G970U1 
● Update: 
Build Number : G970U1UES4ETJ1
Android version : Q(Android 10)
Release Date : 2020-10-20
Security patch level : 2020-10-01

Tested:

• xbox.com / works now with img download
• playstation.com / works now
• twitter.com / works now img 
• nytimes.com / works now img
• reddit.com / works img
• skscans.com / works
• rule34.xxx / works

All of the above appear in default gallery app.
Truthfully I'm not sure if this works nows because of the update to the browser or the phone.... 

Tested the same sites on Firefox Nightly Nightly 201027 17:01 (Build #2015772171) and all of the above works.

Bar 1 exception.
• rule34hentai.net
This site seems to brake in both Firefox versions when you first enter as it tries to give you mobile version but it does not have one so when you enter the page of a video it freezes the page, not the browser just the page as in you can't press anywhere unless you switch to Desktop mode in both versions.

Tried downloading in Firefox Daylight after switching to desktop mode and it works surprisingly with img and some videos not all, but it seems to now not be working in Nightly as the download does not appear in the gallery app.
■ Anyway just gave up with that one lol.

**Comment by Amejia481** [comment]
Thanks @OddEvans, I will take a look as soon as I finish with other bugs that I have for this [sprint](https://github.com/orgs/mozilla-mobile/projects/3)

**Comment by KrasnayaPloshchad** [comment]
What happened if Firefox integrated [storage access framework](https://developer.android.com/guide/topics/providers/document-provider)?

**Comment by softvision-miralobontiu** [comment]
Hi all!
I've tested this issue on the 2/25 Nightly, Beta 87.0.0-beta.1, and RC 86.1.1 with Sony Xperia z5 Premium (Android 7.1.1), and Google Pixel (Android 10), and the downloaded files are added and are visible in the Downloads folder.
I've downloaded .apk, .mp3, .mp4, png, jpg, .gif files.
I'll close this as verified.
